testutil/testnode/full_node.go

```diff
@@ -221,3 +230,15 @@ func DefaultNetwork(t *testing.T, blockTime time.Duration) (cleanup func() error
 		return cleanupGRPC()
 	}, accounts, cctx
 }
+
+func GetFreePort() int {
```

Comment:
Don't we have this function somewhere already

Reply:
yeah on main we do, cherry picked it over from main https://github.com/celestiaorg/celestia-app/pull/1672/commits/4eff397763a333250efeb4146187112ec8b46c88 and then exported to use it in the integration test